site/app/controllers/admin/PlagiarismController.php

```diff
@@ -214,105 +261,108 @@ public function showPlagiarismResult($gradeable_id) {
+            || file_exists(FileUtils::joinPaths($this->core->getConfig()->getSubmittyPath(), "daemon_job_queue", "PROCESSING_lichen__{$semester}__{$course}__{$gradeable_id}.json"))
+        ) {
             $this->core->addErrorMessage("A job is already running for the gradeable. Try again after a while.");
             $this->core->redirect($return_url);
         }
 
-        $prev_gradeable_number = $_POST['prior_term_gradeables_number'];
-        $ignore_submission_number = $_POST['ignore_submission_number'];
-        $version_option = $_POST['version_option'];
-        if ($version_option == "active_version") {
-            $version_option = "active_version";
-        }
-        else {
-            $version_option = "all_version";
-        }
 
-        $file_option = $_POST['file_option'];
-        if ($file_option == "regex_matching_files") {
-            $file_option = "matching_regex";
+        // Upload instructor provided code
+        if ($new_or_edit === "edit" && ($_POST['provided_code_option'] !== "code_provided" || $_FILES['provided_code_file']['tmp_name'] !== "")) {
+            // delete the old provided code
+            $this->deleteExistingProvidedCode($gradeable_id);
         }
-        else {
-            $file_option = "all";
-        }
-        if ($file_option == "matching_regex") {
-            if (isset($_POST['regex_to_select_files']) && $_POST['regex_to_select_files'] !== '') {
-                $regex_for_selecting_files = $_POST['regex_to_select_files'];
+        if ($_POST['provided_code_option'] === "code_provided" && $_FILES['provided_code_file']['tmp_name'] !== "") {
+            // error checking
+            if (empty($_FILES) || !isset($_FILES['provided_code_file']) || !isset($_FILES['provided_code_file']['tmp_name']) || $_FILES['provided_code_file']['tmp_name'] === "") {
+                $this->core->addErrorMessage("Upload failed: Instructor code not provided");
+                $this->core->redirect($return_url);
             }
-            else {
-                $this->core->addErrorMessage("No regex provided for selecting files");
+            // save the code
+            $ret_str = $this->saveNewProvidedCode($_FILES['provided_code_file']['tmp_name'], $_FILES['provided_code_file']['name'], $gradeable_id);
+            if ($ret_str !== "") {
+                $this->core->addErrorMessage("ERROR: could not upload instructor provided code");
                 $this->core->redirect($return_url);
             }
         }
 
+
+        // Version
+        $version_option = $_POST['version_option'];
+        assert($version_option == "active_version" || $version_option == "all_versions");
+
+
+        // Regex
+        // TODO: Can we find a way to validate the regex here to tell the user their regex was invalid before feeding it to Lichen?
+        assert(isset($_POST["regex_dir"]) && isset($_POST["regex_to_select_files"]));
+        $regex_directories = $_POST["regex_dir"];
```

Comment:
I'm pretty sure that the part in the concatenation code that uses the provided regex (around [this line](https://github.com/Submitty/Lichen/blob/0777e2594b0aa418b0c13f6b2cc48ecb8927b1ce/bin/concatenate_all.py#L85)) is only collecting the names of files that exist in each folder (`submissions`, `results`, `checkout`, including files in nested directories) and then filters that list of files based on the regex. If that it the case, then I think breaking out of the submission folder should not be possible

Reply:
My concern was more about Submitty/Lichen#34 which changed how these fields were being used, and looks like @bmcutler raised similar concerns.

This line [https://github.com/Submitty/Lichen/pull/34/files#diff-30ca4b9b6281d4ace1b3c599e62652265d9ecf4c4180139cbb9556cae0e2e205R61] in the changeset looks like it would work alright to jump to a different course if I were to do `../../different_course/submissions` as a regex_dir?

Reply:
I just updated the Lichen PR to check for those three directories.  I believe that it just collects the files and then selects any of the files which match the pattern.  There is no way to match directories unless I am mistaken so I think any concerns have been addressed.